src/game/Update.cpp

```diff
@@ -593,6 +605,69 @@ void Game::resolveDamage() //player walks on a trap
     }
 }
 
+void Game::resolveSnorlax()
+{
+    Item **items = rooms[player.getCurrentRoom()]->getItems();
+    for (int i = 0; i < rooms[player.getCurrentRoom()]->getMaxItems(); i++)
+    {
+        if (items[i] != NULL)
+        {
+            if (items[i]->getYPos() == 6 && items[i]->getXPos() == 6 && (items[i]->getType() == "snorlax") ) // if Snorlax is in starting position
+            {
+                items[i]->setYPos(5);
+                items[i]->setXPos(7);
+            }
+        }
+    }
+}
+
+void Game::usePokeball()
+{
+    Item **items = rooms[player.getCurrentRoom()]->getItems();
+    for (int i = 0; i < rooms[player.getCurrentRoom()]->getMaxItems(); i++)
+    {
+        if (items[i] != NULL)
+        {
+            if (items[i]->getYPos() == 5 && items[i]->getXPos() == 7 && (items[i]->getType() == "snorlax") ) // if Snorlax is in second position
+            {
+                items[i]->setWeight(1);
+                bool pickedUp = player.pickUp(items[i]);
+                if (pickedUp)
+                {
+                    items[i] = NULL;
+                }
+            }
+        }
+    }
+}
+
+void Game::resolveBomb(int y, int x)
+{
+    Item **items = rooms[player.getCurrentRoom()]->getItems();
+    for (int i = 0; i < rooms[player.getCurrentRoom()]->getMaxItems(); i++)
+    {
+        if (items[i] != NULL)
+        { 
+            if ( ((y - 1) == items[i]->getYPos()) && (x == items[i]->getXPos()) && (items[i]->getType() == "softblock") ) // if softblock is above
+            {
+               items[i] = NULL;
+            }
+            else if ( ((y + 1) == items[i]->getYPos()) && (x == items[i]->getXPos()) && (items[i]->getType() == "softblock") ) // if softblock is below
+            {
+              items[i] = NULL;
+            }
+            else if ( (y == items[i]->getYPos()) && ((x-1) == items[i]->getXPos()) && (items[i]->getType() == "softblock") ) // if softblock is to left
+            {
+               items[i] = NULL;
+            }
+            else if ( (y == items[i]->getYPos()) && ((x+1) == items[i]->getXPos()) && (items[i]->getType() == "softblock") ) // if softblock is to right
+            {
+               items[i] = NULL;
+            }

```

Comment:
Setting these to null will cause a memory leak if we don’t first delete what the pointer is pointing to. Keep items[i] = NULL. But just before it delete items[i]